Home directory on alternate volume causes "Failed to write settings to file"

### Summary

<!-- Please insert a one line summary of the issue below -->
User home directories on volumes other than / prevent Zed from performing actions which modify ~/.config/zed/settings.json.

<!-- Include all steps necessary to reproduce from a clean Zed installation. Be verbose -->
Steps to trigger the problem:
1. Create a new APFS volume or attach an external drive.
2. Create a new user whose home directory on a drive other than /
3. Open Zed, do some action which attempts to re-write settings (select model from Assistant, change default keymap, etc).

Actual Behavior:
```
2024-09-12T08:19:33.679273+10:00 [ERROR] Failed to write settings to file "/Users/ext/xxx/.config/zed/settings.json"

Caused by:
0: failed to persist temporary file: Cross-device link (os error 18)
1: Cross-device link (os error 18)
```

Expected Behavior:
Settings file is saved.

Workaround:
Create a symlink for `~/.config/zed/` which points to something on the `/` volume.

Editors note (@notpeter): I have not performed these steps and derived them from comments here:
- https://github.com/zed-industries/zed/issues/16618#issuecomment-2623568239

### Zed Version and System Specs

Zed: v0.150.0 (Zed Preview)
OS: macOS 15.0.0
Memory: 16 GiB
Architecture: x86_64

--- a/crates/fs/src/fs.rs
+++ b/crates/fs/src/fs.rs
@@ -528,14 +528,11 @@ impl Fs for RealFs {
     #[cfg(not(target_os = "windows"))]
     async fn atomic_write(&self, path: PathBuf, data: String) -> Result<()> {
         smol::unblock(move || {
-            let mut tmp_file = if cfg!(any(target_os = "linux", target_os = "freebsd")) {
-                // Use the directory of the destination as temp dir to avoid
-                // invalid cross-device link error, and XDG_CACHE_DIR for fallback.
-                // See https://github.com/zed-industries/zed/pull/8437 for more details.
-                tempfile::NamedTempFile::new_in(path.parent().unwrap_or(paths::temp_dir()))
-            } else {
-                tempfile::NamedTempFile::new()
-            }?;
+            // Use the directory of the destination as temp dir to avoid
+            // invalid cross-device link error, and XDG_CACHE_DIR for fallback.
+            // See https://github.com/zed-industries/zed/pull/8437 for more details.
+            let mut tmp_file =
+                tempfile::NamedTempFile::new_in(path.parent().unwrap_or(paths::temp_dir()))?;
             tmp_file.write_all(data.as_bytes())?;
             tmp_file.persist(path)?;
             anyhow::Ok(())

